worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/AbstractPlayerActor.java

```diff
@@ -520,22 +531,72 @@ public Region getLargestRegion() {
+            case Polygonal2DRegion blockVector3s -> new Polygonal2DRegionSelector(blockVector3s);
+            default -> new CuboidRegionSelector(null, region.getMinimumPoint(), region.getMaximumPoint());
+        };
         selector.setWorld(region.getWorld());
 
         getSession().setRegionSelector(getWorld(), selector);
     }
+
+    @Override
+    public void loadClipboardFromDisk() {
+        if (!clipboardLoading.tryAcquire()) {
+            if (!Fawe.isMainThread()) {
+                try {
+                    clipboardLoading.acquire();
+                    clipboardLoading.release();
+                } catch (InterruptedException e) {
+                    LOGGER.error("Error waiting for clipboard-on-disk loading for player {}", getName(), e);
+                }
+            }
+            return;
+        }
+
+        File file = MainUtil.getFile(
+                Fawe.platform().getDirectory(),
+                Settings.settings().PATHS.CLIPBOARD + File.separator + getUniqueId() + ".bd"
+        );
+        try {
+            Future<?> fut = Fawe.instance().submitUUIDKeyQueuedTask(getUniqueId(), () -> {
+                try {
+                    getSession().loadClipboardFromDisk(file);
+                } catch (FaweClipboardVersionMismatchException e) {
+                    print(e.getComponent());
+                } catch (RuntimeException e) {
+                    print(Caption.of("fawe.error.clipboard.invalid"));
+                    LOGGER.error("Error loading clipboard form disk", e);
+                    print(Caption.of("fawe.error.stacktrace"));
+                    print(Caption.of("fawe.error.clipboard.load.failure"));
+                    print(Caption.of("fawe.error.clipboard.invalid.info", file.getName(), file.length()));
+                    print(Caption.of("fawe.error.stacktrace"));
+                } catch (Exception e) {
+                    print(Caption.of("fawe.error.clipboard.invalid"));
+                    LOGGER.error("Error loading clipboard form disk", e);
+                    print(Caption.of("fawe.error.stacktrace"));
+                    print(Caption.of("fawe.error.no-failure"));
+                    print(Caption.of("fawe.error.clipboard.invalid.info", file.getName(), file.length()));
+                    print(Caption.of("fawe.error.stacktrace"));
+                } finally {
+                    clipboardLoading.release();
+                }
+            });
+            if (Fawe.isMainThread()) {
+                return;
+            }
+            fut.get();
+        } catch (Exception e) {
+            print(Caption.of("fawe.error.clipboard.invalid"));
+            LOGGER.error("Error loading clipboard form disk", e);
+            print(Caption.of("fawe.error.stacktrace"));
+            print(Caption.of("fawe.error.no-failure"));
+            print(Caption.of("fawe.error.clipboard.invalid.info", file.getName(), file.length()));
+            print(Caption.of("fawe.error.stacktrace"));
```

Comment:
Those errors are not really fitting I'd say - the only exception(s) can be thrown by either enqueuing the task or waiting for the completion (e.g. interrupted). I guess a more-generic error would be more fitting, as nothing is wrong with the clipboard itself.